src/dysh/spectra/spectrum.py

```diff
@@ -162,6 +218,9 @@ def _undo_baseline(self):
         if self._baseline_model is None:
             return
         if self._subtracted:
+            if self._normalized:
+                warnings.warn("Cannot undo previously normalized baseline subtraction")
```

Comment:
Don't you already undo it in line 208?

Reply:
It seems the _undo_baseline() function can be removed, as the removal is indeed done inside of baseline(). I doesn't hurt, since the self._subtracted was properly set. So perhaps somebody might want to not subtract, do some inspection, and later decide to call _undo_baseline() manually???

Must admit not having tested this.

Reply:
We want to keep _undo_baseline, in fact it should be promoted by removing the underscore.  It is the only undo we kept in the various requirements iterations.   It adds the model back in if subtracted and resets model to None.   You can remove lines 221/222 if self._normalized.